frontend/src/static/js/v_authorship.js

```diff
@@ -316,4 +316,7 @@ window.vAuthorship = {
     this.initiate();
     this.setInfoHash();
   },
+  components: {
+    v_segment: window.vSegment,
```

Comment:
Not sure if this is necessary.

Reply:
Yes I do think it is necessary as the segment is a part of the authorship page. Also, it opens a path to pass data (via props) from the authorship component to the segment component.

Reply:
Correct me if I am wrong because I am still new to Vue.

Reply:
I think you are correct, never mind. The way we use Vue in RepoSense is a bit different so I am not sure also.

Reply:
@ongspxm for your input

Reply:
yeah, the `window.vSegment` is an JS object, not an actual vue object. Legacy problems, cos last time we not big enough to put everything as Vue component. You can ask for permission to rewrite the frontend, and start shifting everything into their own component. cos it is getting quite big now.